### PR TITLE
Add pool naming from sample sheets for Illumina

### PR DESCRIPTION
**Pull request by apmasell**

### CODE REVIEW

File: runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultIllumina.java [hunk 1]
```diff
@@ -91,6 +99,18 @@ public NotificationDto process(File runDirectory, TimeZone tz) throws IOExceptio
...
+        .orElse(Stream.empty())//
+        .filter(LimsUtils.rejectUntil(line -> line.startsWith("Sample_ID,")))//
+        .map(COMMA::split)//
+        .map(Pair.number(1))
+        .filter(pair -> pair.getValue().length > 0)//
+        .collect(Collectors.toMap(Entry::getKey, e -> e.getValue()[0])));
+
     // The Illumina library can't distinguish between a failed run and one that either finished or is still going. Scan the logs, if
     // available to determine if the run failed.
     File rtaLogDir = new File(runDirectory, "/Data/RTALogs");
```
Comment (sqwci) [inline runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultIllumina.java]: MINOR Refactor your code to get this URI from a customizable parameter. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1075)

File: runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultIllumina.java [hunk 2]
```diff
@@ -66,6 +72,8 @@ public DefaultIllumina(Builder builder) {
     return Arrays.stream(root.listFiles(f -> f.isDirectory() && !f.getName().equals("Instrument")));
   }
 
+  private static final Pattern COMMA = Pattern.compile(",");
+
   @Override
   public NotificationDto process(File runDirectory, TimeZone tz) throws IOException {
     // Call the C++ program to do the real work and write a notification DTO to standard output. The C++ object has no direct binding to the
```
Comment (sqwci) [inline runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultIllumina.java]: MAJOR This block of commented-out lines of code should be removed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

**Comment by sqwci** [comment]
SonarQube analysis reported 11 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![CRITICAL][CRITICAL] 3 critical
* ![MAJOR][MAJOR] 5 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### 9 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![BLOCKER][BLOCKER] [LimsUtils.java#L250](https://github.com/TGAC/miso-lims/blob/9a0d7d344cd97b29e3d4dc6c8c0df702d8ef64d5/core/src/main/java/uk/ac/bbsrc/tgac/miso/core/util/LimsUtils.java#L250): Use try-with-resources or close this "BufferedOutputStream" in a "finally" clause. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2095)
1. ![CRITICAL][CRITICAL] [IlluminaNotificationDto.java#L80](https://github.com/TGAC/miso-lims/blob/9a0d7d344cd97b29e3d4dc6c8c0df702d8ef64d5/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/IlluminaNotificationDto.java#L80): Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
1. ![CRITICAL][CRITICAL] [NotificationDto.java#L134](https://github.com/TGAC/miso-lims/blob/9a0d7d344cd97b29e3d4dc6c8c0df702d8ef64d5/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/NotificationDto.java#L134): Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
1. ![CRITICAL][CRITICAL] [PacBioNotificationDto.java#L41](https://github.com/TGAC/miso-lims/blob/9a0d7d344cd97b29e3d4dc6c8c0df702d8ef64d5/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/PacBioNotificationDto.java#L41): Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
1. ![MAJOR][MAJOR] [LimsUtils.java#L769](https://github.com/TGAC/miso-lims/blob/9a0d7d344cd97b29e3d4dc6c8c0df702d8ef64d5/core/src/main/java/uk/ac/bbsrc/tgac/miso/core/util/LimsUtils.java#L769): Remove this unused method parameter "timezone". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [DefaultIllumina.java#L47](https://github.com/TGAC/miso-lims/blob/9a0d7d344cd97b29e3d4dc6c8c0df702d8ef64d5/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultIllumina.java#L47): Remove this unused method parameter "parameters". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [DefaultIllumina.java#L58](https://github.com/TGAC/miso-lims/blob/9a0d7d344cd97b29e3d4dc6c8c0df702d8ef64d5/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultIllumina.java#L58): Use the built-in formatting to construct this argument. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2629)
1. ![MAJOR][MAJOR] [DefaultIllumina.java#L91](https://github.com/TGAC/miso-lims/blob/9a0d7d344cd97b29e3d4dc6c8c0df702d8ef64d5/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultIllumina.java#L91): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2142)
1. ![MINOR][MINOR] [DefaultIllumina.java#L54](https://github.com/TGAC/miso-lims/blob/9a0d7d344cd97b29e3d4dc6c8c0df702d8ef64d5/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultIllumina.java#L54): Replace this "switch" statement by "if" statements to increase readability. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1301)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultIllumina.java [hunk 3]
```diff
@@ -91,6 +99,18 @@ public NotificationDto process(File runDirectory, TimeZone tz) throws IOExceptio
...
+    dto.setPoolNames(Optional.of(new File(runDirectory, "SampleSheet.csv"))//
+        .filter(File::canRead)//
+        .map(File::toPath)
+        .map(WhineyFunction.log(log, Files::lines))//
+        .orElse(Stream.empty())//
+        .filter(LimsUtils.rejectUntil(line -> line.startsWith("Sample_ID,")))//
+        .map(COMMA::split)//
+        .map(Pair.number(1))
+        .filter(pair -> pair.getValue().length > 0)//
+        .collect(Collectors.toMap(Entry::getKey, e -> e.getValue()[0])));
```
Comment (djcooke) [inline runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultIllumina.java]: Could easily be moved into a separate method and unit tested

**Pull request merged**